exp-android-sdk/src/main/java/com/scala/exp/android/sdk/model/ContentNode.java

```diff
@@ -83,9 +109,12 @@ public String getUrl(){
      */
     public String getVariantUrl(String name){
         if(Utils.CONTENT_TYPES.FILE == this.subtype && hasVariant(name)){
-            return getUrl() + VARIANT + name;
+            String rt = AppSingleton.getInstance().getUser().getRestrictedToken();
+            return new StringBuilder(getUrl())
+                    .append("?").append(VARIANT).append(name)
+                    .append("&").append(RT).append(rt).toString();
         }
-        return "";
+        return null;
```

Comment:
yeah I agree well the small difference that if you have nullpointer in JAVA the app will crash I think in SWIFT the app still runs, but we should have something similar for all the SDK since is a particular method